### ISSUE TITLE
Slate commands not working - ENOENT: no such file or directory, scandir

### ISSUE THREAD

**Issue by mattkersner** [Shopify/slate#528]
### Problem

Getting error for every slate command:

```
fs.js:904
  return binding.readdir(pathModule._makeLong(path), options.encoding);
                 ^

Error: ENOENT: no such file or directory, scandir '/Users/mattkersner/Desktop/Code/figs-test-mk/node_modules/@shopify/slate-tools/lib/commands'
    at fs.readdirSync (fs.js:904:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/@shopify/slate/lib/index.js:93:23)
    at Module._compile (module.js:643:30)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Function.Module.runMain (module.js:684:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
```

### Replication steps

Ran this from command line:
1. slate theme my-theme
2. cd my-theme
3. yarn (have also tried npm install)
4. set up config.yml
5. slate zip - throws error
slate watch -e development - throws error
slate build - throws error
even slate -v throws the error

### More Information

I find it odd that it's looking in 
/my-theme/node_modules/@shopify/slate-tools/lib/commands

when I could see the commands folder in 
/my-theme/node_modules/@shopify/slate-tools/cli/commands

not sure if that's pertinent 

Edit: 

##step-by-step images, starting a new theme from scratch: 
<img width="568" alt="screen shot 2018-05-03 at 10 23 56 pm" src="https://user-images.githubusercontent.com/23509139/39610788-be00871a-4f20-11e8-882a-bd20f5c4675f.png">
<img width="568" alt="screen shot 2018-05-03 at 10 24 05 pm" src="https://user-images.githubusercontent.com/23509139/39610791-c3d56a34-4f20-11e8-908d-e902593fabae.png">
<img width="564" alt="screen shot 2018-05-03 at 10 26 17 pm" src="https://user-images.githubusercontent.com/23509139/39610838-0c48d56c-4f21-11e8-9f49-c16c6c8ecd45.png">
##Debugging
<img width="570" alt="screen shot 2018-05-03 at 10 27 20 pm" src="https://user-images.githubusercontent.com/23509139/39610872-5130137a-4f21-11e8-9ae9-fd1c11d7dfd2.png">
<img width="566" alt="screen shot 2018-05-03 at 10 27 31 pm" src="https://user-images.githubusercontent.com/23509139/39610876-577f093e-4f21-11e8-85ef-23d871a0d3e5.png">
<img width="564" alt="screen shot 2018-05-03 at 10 27 40 pm" src="https://user-images.githubusercontent.com/23509139/39610885-6198f1e6-4f21-11e8-928d-17aa9aa42c8d.png">
<img width="563" alt="screen shot 2018-05-03 at 10 29 16 pm" src="https://user-images.githubusercontent.com/23509139/39610905-81a5dc4c-4f21-11e8-9ed6-e62dff6eab35.png">
<img width="564" alt="screen shot 2018-05-03 at 10 29 23 pm" src="https://user-images.githubusercontent.com/23509139/39610909-88e8e166-4f21-11e8-9d35-6327064e9b9c.png">

**Comment by mattkersner** [comment]
also tried uninstalling slate globally, and it never uninstalls. 

I run npm uninstall -g @shopify/slate, it goes through the process and tells me it removed 158 packages. Then I type in slate -v and it comes up 0.13.0.

Something tells me this is the heart of the problem, as it could be conflict of some sorts? But how am I still able to see the version when I uninstall it globally?

**Comment by mattkersner** [comment]
Still trying to debug this and desperately searching for help. This literally has me not able to do any Slate development. I do thing it's weird that I could use both the yarn create slate-theme command (slate 1.0) and the slate theme command (slate 0.13). Whenever I try slate -v, I always get 0.13 though, am yet to see slate 1.0 when checking. Could that be the conflict?

**Comment by t-kelly** [comment]
@mattkersner please don't spam old, unrelated issues with links to your own. This pollutes our repo history, creates unneeded notification, and ultimately slows down our response to your actual issue.

If you're starting with Slate for the first time, I would recommend you follow Slate v1's [Getting Started guide](https://github.com/Shopify/slate/wiki/Getting-Started) for how to get up and running. Slate v0 and Slate v1 are major version bumps, and function very differently so don't expect the documentation from one version to apply to the other.

It is normal that the default Starter Theme looks like the styles are broken. That is on purpose as style are project specific and vary greatly between project, so we chose not to include them. I would recommend that you try adding your own styles to confirm if they are, in fact, broken or not.

**Comment by mattkersner** [comment]
My apologies, I'm fairly new to creating Github issues, will not do that again. 

I followed the Slate V1 getting started guide (the yarn create commands), and that seems to work ok. My issue is with v0, that is what is throwing those errors and not even allowing me to check slate -v without getting that error. Question; is there a slate migrate feature in v1? I did not see any mention of that kind of command in the documentation. I do want to do development on a pre-existing theme (Debut) for testing purposes. 

Also, I am really curious as to why I am getting these errors and can no longer use Slate v0. Is it possible there is a conflict with the slate versions? Whenever I run slate -v globally I always get 0.13.0 

I checked my /usr/local/bin file and can confirm that there is a slate bin in there. I can also confirm that there is a @shopify folder in my /usr/local/lib/node_modules folder. 

node -v : 8.6.0
npm -v : 6.0.0
slate -v : 0.13.0

**Comment by t-kelly** [comment]
Due to the advanced level of bundling and compilation Slate v1 does on build, it is not possible for us to implement a migrate command.

I'll give Slate v0 another run through as it should still be working. Will keep you posted on what I find.

**Comment by hisyam** [comment]
I got this error when running any slate v0 commands:

$ slate -v

node[4498]: ../src/node_contextify.cc:631:static void node::contextify::ContextifyScript::New(const FunctionCallbackInfo<v8::Value> &): Assertion `args[1]->IsString()' failed.
 1: node::Abort() [/usr/local/bin/node]
 2: node::Assert(char const* const (*) [4]) [/usr/local/bin/node]
 3: node::contextify::ContextifyScript::New(v8::FunctionCallbackInfo<v8::Value> const&) [/usr/local/bin/node]
 4: v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo*) [/usr/local/bin/node]
 5: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<true>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) [/usr/local/bin/node]
 6: v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) [/usr/local/bin/node]
 7: 0xca79f00427d
Abort trap: 6

---
I'm using :
node v10.0.0
npm 5.6.0
slate version... 0.13.0 ...I guess?

**Comment by mattkersner** [comment]
@t-kelly, when running Slate migrate, a user noted that the package.json file created had a dev dependency of @shopify/slate-tools: 1.0.0-beta.1. Once I changed that to @shopify/slate-tools: 0.13.0 and ran npm install, everything worked correctly.

**Comment by t-kelly** [comment]
@mattkersner would you mind creating a new issue for that? I'll jump on it ASAP. Thanks for digging into this and finding the source of the problem! Great help.

**Comment by mattkersner** [comment]
No problem! Thank you!

**Comment by nemoeric** [comment]
Awesome @mattkersner 
Lost my working session on that... ! 
Looking forward on the slate migrate cmd for slate V1 @t-kelly  !

**Comment by t-kelly** [comment]
This thread is has been blown out of scope is too hard to follow. Locking it. Please open a new issue something here needs to further addressed.
